src/plugins/intel_npu/src/plugin/npuw/partitioning/patterns/opt.cpp

```diff
@@ -185,13 +185,88 @@ DQMatMulCWi::DQMatMulCWi() {
             for (auto&& r : mm_readers) {
                 r.replace_source_output(matched_node_cvtm);
             }
+            return true;  // root has changed
         }
-
-        return true;  // root has changed
+        return false;  // root hasn't changed
     };
     register_matcher(std::make_shared<opp::Matcher>(qmm, "OptDQMatMulCWi"), std::move(callback));
 }
 
+// FROM:
+//     ???(Act) ------------------------>
+//     Param(W) -> to(f32) -> Multiply -> MatMul
+//     Param(S) ------------>
+//
+// TO:
+//     ???(Act) -> to(f32) ->
+//     Param(W) -> to(f32) -> MatMul -> Multiply
+//     Param(S) -> Reshape ----------->
+//
```

Comment:
Instead of this, I'd recommend you to extend the existing pattern. You can mark `to_f32` as `optional` there and handle both cases in just one pattern

Reply:
Done